daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1544,6 +1551,13 @@ private[lf] object SBuiltin {
           machine.disclosureTable.contractIdByKey.get(gkey.hash) match {
             case Some(coid) =>
               val vcoid = coid.value
+              val disclosedContract = machine.disclosureTable.disclosedContractByKey.getOrElse(
+                gkey.hash,
+                crash(
```

Comment:
By construction, the disclosure table should have a contract key to disclosed contract entry for every defined contract key, hence we crash if this is not the case